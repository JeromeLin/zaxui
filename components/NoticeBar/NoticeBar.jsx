import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import Message from '../Message';
import Icon from '../Icon';

class NoticeBar extends PureComponent {

  constructor(props) {
    super(props);
    this.state = {
      offset: 0,
    };
  }

  componentDidMount() {
    const distance = this.content.offsetWidth - this.wrapper.offsetWidth;
    if (distance <= 0) return;

    setInterval(() => {
      let { offset } = this.state;
      offset = (offset < -distance)
        ? 0
        : (offset - 1);

      this.setState({
        offset,
      });
    }, 50);
  }

  render() {
    const { prefixCls, children, ...others } = this.props;

    return (
      <Message {...others} size="lg">
        <div className={prefixCls} ref={(ele) => { this.wrapper = ele; }}>
          <div className={`${prefixCls}-body`} ref={(ele) => { this.content = ele; }} style={{ left: this.state.offset }}>{children}</div>
        </div>
      </Message>
    );
  }
}

NoticeBar.propTypes = {
  prefixCls: PropTypes.string,
  className: PropTypes.string,
  theme: Message.propTypes.theme,
  mode: Message.propTypes.mode,
  icon: Message.propTypes.icon,
};

NoticeBar.defaultProps = {
  prefixCls: 'za-noticebar',
  className: null,
  theme: 'warning',
  mode: null,
  icon: <Icon type="broadcast" />,
};

export default NoticeBar;
