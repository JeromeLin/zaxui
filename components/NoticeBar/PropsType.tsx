import { MessageProps } from '../Message/PropsType';

export interface NoticeBarProps extends MessageProps {
  autoscroll?: boolean;
}
