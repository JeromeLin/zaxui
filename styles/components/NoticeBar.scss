$prefixCls: za-noticebar;

.#{$prefixCls} {
  width: 100%;
  height: r(24);
  position: relative;
  overflow: hidden;

  &-body {
    position: absolute;
    left: 0;
    top: 2px;
    white-space: nowrap;
  }
}
