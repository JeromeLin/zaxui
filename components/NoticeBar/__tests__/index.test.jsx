import React from 'react';
import { render } from 'enzyme';
import toJson from 'enzyme-to-json';
import NoticeBar from '../index';

describe('NoticeBar', () => {
  it('renders correctly', () => {
    const wrapper = render(<NoticeBar>foo</NoticeBar>);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
