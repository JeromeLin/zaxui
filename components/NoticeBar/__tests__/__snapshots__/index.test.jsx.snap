// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NoticeBar renders correctly 1`] = `
<div
  class="za-message theme-warning size-lg"
>
  <div
    class="za-message-header"
  >
    <div
      class="za-message-icon"
    >
      <i
        class="za-icon za-icon-broadcast"
      />
    </div>
  </div>
  <div
    class="za-message-body"
  >
    <div
      class="za-noticebar"
    >
      <div
        class="za-noticebar-body"
        style="left:0;"
      >
        foo
      </div>
    </div>
  </div>
</div>
`;
